frontend/src/summary_charts.pug

```diff
@@ -61,35 +73,35 @@
         .summary-chart__title--contribution.mini-font [{{ getUserTotalContribution(user) }} lines]
         a(
           v-if="filterGroupSelection !== 'groupByRepos'",
-          v-bind:href="getRepoLink(repo[i])", target="_blank",
+          v-bind:href="getRepoLink(repo[j])", target="_blank",
           title="click to view repo"
         )
           font-awesome-icon.icon-button(:icon="['fab', 'github']")
         a(
           v-if="filterGroupSelection !== 'groupByAuthors'",
-          v-bind:href="getAuthorProfileLink(repo[i].name)", target="_blank",
+          v-bind:href="getAuthorProfileLink(repo[j].name)", target="_blank",
           title="click to view author's profile"
         )
           font-awesome-icon.icon-button(icon="user")
         a(
           title="click to view author's code",
           onclick="deactivateAllOverlays()",
-          v-on:click="openTabAuthorship(user, repo, i)"
+          v-on:click="openTabAuthorship(user, repo, j, mergedGroupsMap[getGroupName(repo)])"
```

Comment:
Moreover, `mergedGroupsMap[getGroupName(repo)])` is used so many time, is it able to set a variable inside the `repo` class to represent for it is a merged Group or not?

The logic will be clearer as it only needs to check the attribute of the repo object. 

@jamessspanggg Your idea?

Reply:
I believe currently we do not have a `repo` class. Right now a `repo` is a list of groups. Changing it to a class will require some refactoring. I am not sure if we should change so much just for this feature. But I agree that it can be improved.

Reply:
> Moreover, `mergedGroupsMap[getGroupName(repo)])` is used so many time, is it able to set a variable inside the `repo` class to represent for it is a merged Group or not?
> 
> The logic will be clearer as it only needs to check the attribute of the repo object.
> 
> @jamessspanggg Your idea?

yes I believe it will be much clearer that way. But the way our current `repo` (a list) is structured, it makes it harder/more complex to achieve certain features like this PR. 

I believe we need to re-think and consider changing the structure of our `repo`, it may incur a lot of changes, but it will definitely make things much easier in the long run. Maybe in a separate PR.